File: Ser2023/Company/Company.Web/Modules/Company/Country/CountryColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Company.Country")]
[BasedOnRow(typeof(CountryRow), CheckNames = true)]
public class CountryColumns
{
    [EditLink, Width(80), DisplayName("Db.Shared.RecordId"), AlignRight]
    public string CountryCode { get; set; }
    [Width(200)]
    public string CountryName { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/CountryColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CountryRow } from "./CountryRow";

export interface CountryColumns {
    CountryCode: Column<CountryRow>;
    CountryName: Column<CountryRow>;
}

export class CountryColumns extends ColumnsBase<CountryRow> {
    static readonly columnsKey = 'Company.Country';
    static readonly Fields = fieldsProxy<CountryColumns>();
}